core/src/main/java/jenkins/security/ApiTokenStore.java

```diff
@@ -218,89 +178,41 @@ private void addLegacyToken(@Nonnull Secret legacyToken) {
         }
         return digest.digest(tokenBytes);
     }
-
-    private @Nonnull String generatePrefix() {
-        int i = 0;
-        boolean unique;
-        String currentPrefix;
-        
-        do {
-            currentPrefix = generateRandomPrefix();
-            unique = !prefixToTokenList.containsKey(currentPrefix);
-            i++;
-        } while (i < MAX_ATTEMPTS && !unique);
-        
-        Level logLevel = Level.FINE;
-        if (i == MAX_ATTEMPTS) {
-            logLevel = Level.WARNING;
-        }
-        LOGGER.log(logLevel, "Prefix generated after {0}/{1} attempts", new Object[]{i, MAX_ATTEMPTS});
-        
-        return currentPrefix;
-    }
-    
-    /**
-     * Generate random 3-hex-character
-     */
-    @SuppressFBWarnings("NP_NONNULL_RETURN_VIOLATION")
-    private @Nonnull String generateRandomPrefix() {
-        int prefixInteger = RANDOM.nextInt(4096);
-        
-        String prefixString = Integer.toHexString(prefixInteger);
-        return StringUtils.leftPad(prefixString, 3, '0');
-    }
     
     public synchronized boolean doesContainToken(@Nonnull String token) {
-        String prefixToSearch;
         String plainToken;
-        
         if (isLegacyToken(token)) {
-            prefixToSearch = LEGACY_PREFIX;
             plainToken = token;
         } else {
-            prefixToSearch = getPrefixOfToken(token);
             plainToken = getHashOfToken(token);
         }
         
-        return searchMatchUsingPrefix(prefixToSearch, plainToken);
+        return searchMatch(plainToken);
     }
     
     private boolean isLegacyToken(String token) {
         return token.length() != TOKEN_LENGTH_V2;
     }
     
     /**
-     * [1: version][3: prefix][32: real token]
-     * ^^^^^^^^^^^^---------------------------
+     * [2: version][32: real token]
+     * ^^^^^^^^^^^^----------------
      */
     private String getVersionOfToken(String token) {
-        return String.valueOf(token.charAt(0));
-    }
-    
-    /**
-     * [1: version][3: prefix][32: real token]
-     * ------------^^^^^^^^^^^----------------
-     */
-    private String getPrefixOfToken(String token) {
-        return token.substring(1, 4);
+        return token.substring(0, 1);
```

Comment:
`endIndex` is exclusive, so this is just the first character. You want `substring(0, 2)`.